plugins/generator-1.18.2/forge-1.18.2/templates/elementinits/fuels.java.ftl

```diff
@@ -42,9 +43,14 @@ package ${package}.init;
 
 	@SubscribeEvent
 	public static void furnaceFuelBurnTimeEvent(FurnaceFuelBurnTimeEvent event) {
-		<#list fuels as fuel>
-		<#if fuel?index == 0>if<#else>else if</#if>(event.getItemStack().getItem() == ${mappedMCItemToItem(fuel.block)})
-			event.setBurnTime(${fuel.power});
+		ItemStack itemstack = event.getItemStack();
+		<#list itemextensions as extension>
+            <#if extension.enableFuel>
+                if (event.getItemStack().getItem() == ${mappedMCItemToItem(extension.item)})
+                    <#if hasReturnValueOf(extension.fuelSuccessCondition, "logic")>if (<@procedureOBJToConditionCode extension.fuelSuccessCondition/>)</#if>

```

Comment:
`hasReturnValueOf` can be replaced with `hasProcedure` because the boolean value isn't optional (same for the 1.16 template)